shlyapa/config.py

```diff
@@ -0,0 +1,76 @@
+import yaml
+from enum import Enum
+
+default_config_game_filepath = "config/game.yaml"
+
+
+class Config(object):
+    class TypeGame(Enum):
+        ORIGINAL = 0,  # fixed pairs
+        AVA = 1,       # All Vs All is unfair variant game "all vs all" shlyapa move clear on circle
+        AVAF = 2       # All Vs All Fair is more fair variant game "all vs all" shlyapa move not clear on circle
+
+    def __init__(self, type=None, number_players=None, number_words=None, number_tours=3, is_last_turn_in_tour_divisible=True):
+        self.type = None
+        self.number_players = None
+        self.number_words = None
+        self.number_tours = None
+        self.is_last_turn_in_tour_divisible = None
+
+        self.set_type(type)
+        self.set_number_players(number_players)
+        self.set_number_words(number_words)
+        self.set_number_tours(number_tours)
+        self.set_is_last_turn_in_tour_divisible(is_last_turn_in_tour_divisible)
+
+    def set_from_file(self, config_filepath=default_config_game_filepath):
+        with open(config_filepath) as f:
+            values = yaml.safe_load(f)
+            self.set_type(values["type"])
+            self.set_number_players(values["number_players"])
+            self.set_number_words(values["number_words"])
+            self.set_number_tours(values["number_tours"])
+            self.set_is_last_turn_in_tour_divisible(values["is_last_turn_in_tour_divisible"])
+
+    def set_type(self, type):
+        if isinstance(type, Config.TypeGame):
+            self.type = type
+        elif isinstance(type, str):
+            if type == "ORIGINAL":
+                self.type = Config.TypeGame.ORIGINAL
+            elif type == "AVA":
+                self.type = Config.TypeGame.AVA
+            elif type == "AVAF":
+                self.type = Config.TypeGame.AVAF
```

Comment:
Simplify, using classes as IDs:
```suggestion
        if isinstance(type, NextPairAlg):
            self.type = type
        elif isinstance(type, str):
            if type in ('ORIGINAL', 'AVA', 'AVAF'):
                self.type = getattr(sys.modules[__name__], type.title())
```
do not forget `import sys`
and then instead of 
`self.type == Config.TypeGame.ORIGINAL` use just a class-name 
`self.type == Original`